doc/api/esm.md

```diff
@@ -206,16 +206,15 @@ a full path including extension: `"./index.mjs"`, not `"./index"`.
 If the `package.json` `"type"` field is omitted, a `.js` file in `"main"` will
 be interpreted as CommonJS.
 
-> Currently a package can define _either_ a CommonJS entry point **or** an ES
-> module entry point; there is no way to specify separate entry points for
-> CommonJS and ES module usage. This means that a package entry point can be
-> included via `require` or via `import` but not both.
->
-> Such a limitation makes it difficult for packages to support both new versions
-> of Node.js that understand ES modules and older versions of Node.js that
-> understand only CommonJS. There is work ongoing to remove this limitation, and
-> it will very likely entail changes to the behavior of `"main"` as defined
-> here.
+The `"main"` field can point to exactly one file, regardless of whether the
+package is referenced via `require` (in a CommonJS context) or `import` (in an
```

Comment:
The bracketed clarifications seem a bit ambiguous about whether it's the context of the caller or the callee module. I think we're trying to make a statement about the callee module. And it's not true that `import` implies either the caller or the callee (if we consider dynamic `import()` will be in ESM context. I am finding this hard to word accurately and unambiguously. Maybe something like:

>The `"main"` field can point to exactly one file, regardless of whether the package is referenced via `require` (which will load the file as CommonJS) or `import` (which allows Node's loading rules to decide whether to load the file as either an ES module or as CommonJS).

Reply:
I think the original is more correct. It is about which loader / module graph the module may be part of, not about the interpretation of the file contents. E.g. in a `require` context, it may be any number of file formats supported via `require._extensions` and in an `import` context it may be any content type supported by the `import` loader system.

The main point here is that it can only point to exactly one file on disk and it may only be instantiated/interpreted at most once: Either as part of the `require` module graph or as part of the `import` module graph (with the additional note that the `import` graph may point to nodes in the `require` graph).